docs/dev/RELEASE.md

```diff
@@ -2,80 +2,64 @@
-   to be updated, but you may want to increment them further,
-   use judgement. The last section is in "ymmdd" format, where "y" is
-   the number of years since 2018 and "mmdd" the month and day portions
-   of the release date (e.g., a release on January 12th, 2022 would be
-   x.y.z.40112). Edit the internal version number in 
-  `Source/Directory.Build.Props`.
-
-   Put the public version number in place of the "Upcoming" header in
-   `RELEASE_NOTES.md`, and add a new "Upcoming" header above it.
-
-   Push and cut a PR, get it approved, and squash and merge those
-   changes to the `master` branch.
-
-3. Kick off the deep test suite by navigating to
-   https://github.com/dafny-lang/dafny/actions/workflows/deep-tests.yml,
-   clicking the "Run workflow" dropdown, ensuring `master` is selected,
-   and clicking the "Run workflow" button. The automation for releasing
-   below will check for a run of this workflow on the exact commit
-   to release.
-
-4. Create a fresh clone of the repo locally, making sure the latest commit
-   is the squashed commit you just merged and tested, and push the tag for this release.
-
-   ```
-   git clone git@github.com:dafny-lang/dafny.git dafny-for-tagging
-   cd dafny-for-tagging
-   git tag v<$VER>
-   git push origin v<$VER>
-   ```
-
-5. A GitHub action will automatically run in reaction to the tag being pushed,
-   which will build the artifacts, upload them to NuGet, build the reference manual,
-   and then create a draft GitHub release.
-   You can find and watch the progress of this workflow at
-   https://github.com/dafny-lang/dafny/actions.
-
-6. Once the action completes, you should find the draft release at
-   https://github.com/dafny-lang/dafny/releases. Edit the release body
-   to add in the release notes from `RELEASE_NOTES.md`.
-   Also check the box to create a new discussion based on
-   the release, if this is not a pre-release.
-
-7. Push the "Publish" button. This will trigger yet another workflow
-   that will download the published artifacts (from both GitHub and NuGet)
-   and run a smoke test on multiple platforms. 
-   Again you can watch for this workflow at
-   https://github.com/dafny-lang/dafny/actions.
-
-8. If preparing a pre-release, stop here, as
-   the following steps declare the release as the latest version, which
-   is not the intention.
-
-9. If something goes wrong, delete the tag and release in GitHub, fix the
-   problem and try again.
-
-10. Update the Homebrew formula for Dafny (see below).
-    Note that it is fine to leave this for the next day,
-    and other members of the community may update the formula
-    in the meantime anyway.
-
-11. Announce the new release to the world.
+0. Ensure that you are in a clean and up-to-date repository, with the `master`
+   branch checked out and up-to-date and no uncommitted changes.
+
+1. Select a version number `$VER` (e.g., "3.0.0" or "3.0.0-alpha"), then run
```

Comment:
Could we have the script infer the correct next version number based on whether there are any new `feat` or `feat!` news items? Or at least verify you selected the right version number.

Reply:
Sure, do you think it's worth it given that we plan to revise our version policy soon?

Reply:
I'm afraid we would loose the ability to bump the major version number with such an auto-detect 😮 
My best suggestion is that:
- The script detect feat or feat! extensions
- If there are, it suggests to increment the minor version number, and if the user presses ENTER, this is it.
- If there are none (only fixes), it suggests to increment the patch version number, and if the user presses ENTER, this is it
- But if the user decides to put another version number, it takes it and continue.

Reply:
This is a lot of work for saving the user from typing 3 numbers ^^  I vote for leaving this for future improvement